### PR TITLE
docs: why GuestOS deps are required

### PR DESCRIPTION
**Pull request by rumenov**

### CODE REVIEW

File: ic-os/guestos/defs.bzl [hunk 1]
```diff
@@ -28,25 +28,45 @@ def image_deps(mode, malicious = False):
         "bootfs": {},
         "rootfs": {
             # additional files to install
+            # TODO runtime
```
Comment (ulan) [inline ic-os/guestos/defs.bzl:31]: Need for the canister sandboxing to work.

File: ic-os/guestos/defs.bzl [hunk 2]
```diff
@@ -28,25 +28,45 @@ def image_deps(mode, malicious = False):
...
             "//publish/binaries:canister_sandbox": "/opt/ic/bin/canister_sandbox:0755",
+            # TODO runtime
```
Comment (ulan) [inline ic-os/guestos/defs.bzl:33]: Need for the Wasm compilation sandboxing to work.

File: ic-os/guestos/defs.bzl [hunk 3]
```diff
@@ -28,25 +28,45 @@ def image_deps(mode, malicious = False):
...
             "//publish/binaries:orchestrator": "/opt/ic/bin/orchestrator:0755",
+            # API Boundary Node process. The same GuestOS is used both for the replica and API boundary nodes.
             "//publish/binaries:ic-boundary-tls": "/opt/ic/bin/ic-boundary:0755",
+            # Main protocol binary
             ("//publish/malicious:replica" if malicious else "//publish/binaries:replica"): "/opt/ic/bin/replica:0755",  # Install the malicious replica if set
+            # TODO DRE
             "//publish/binaries:metrics-proxy": "/opt/ic/bin/metrics-proxy:0755",
+            # TODO node
             "//publish/binaries:nft-exporter": "/opt/ic/bin/nft-exporter:0755",
+            # TODO runtime
```
Comment (ulan) [inline ic-os/guestos/defs.bzl:61]: Need for the canister/compilation sandboxing to work.

File: ic-os/guestos/defs.bzl [hunk 4]
```diff
@@ -28,28 +28,40 @@ def image_deps(mode, malicious = False):
...
+            # TODO consensus
             "//publish/binaries:ic-recovery": "/opt/ic/bin/ic-recovery:0755",
-            "//publish/binaries:orchestrator": "/opt/ic/bin/orchestrator:0755",
-            "//publish/binaries:ic-boundary-tls": "/opt/ic/bin/ic-boundary:0755",
-            ("//publish/malicious:replica" if malicious else "//publish/binaries:replica"): "/opt/ic/bin/replica:0755",  # Install the malicious replica if set
+            # TODO crypto, node
+            "//publish/binaries:fstrim_tool": "/opt/ic/bin/fstrim_tool:0755",
+            # TODO node
+            "//publish/binaries:guestos_tool": "/opt/ic/bin/guestos_tool:0755",
+            # TODO dre
```
Comment (sasa-tomic) [inline ic-os/guestos/defs.bzl:47]:
```suggestion
            # Proxies, filters, and serves public node metrics
```

File: ic-os/guestos/defs.bzl [hunk 5]
```diff
@@ -28,28 +28,40 @@ def image_deps(mode, malicious = False):
...
             "//publish/binaries:state-tool": "/opt/ic/bin/state-tool:0755",
+            # TODO node
             "//publish/binaries:vsock_guest": "/opt/ic/bin/vsock_guest:0755",
+            # TODO node
             "//cpp:infogetty": "/opt/ic/bin/infogetty:0755",
+            # TODO node
             "//cpp:prestorecon": "/opt/ic/bin/prestorecon:0755",
 
             # additional libraries to install
+            # TODO dre
```
Comment (sasa-tomic) [inline ic-os/guestos/defs.bzl:61]: Allows referring to the guest IPv6 by name `guestos` from host, and host as `hostos` from guest.

**Review by sasa-tomic — APPROVED** [review]

File: ic-os/guestos/defs.bzl [hunk 6]
```diff
@@ -28,29 +28,29 @@ def image_deps(mode, malicious = False):
...
+            "//publish/binaries:compiler_sandbox": "/opt/ic/bin/compiler_sandbox:0755",  # Need for the Wasm compilation sandboxing to work.
+            "//publish/binaries:sandbox_launcher": "/opt/ic/bin/sandbox_launcher:0755",  # Need for the canister/compilation sandboxing to work.
+            "//publish/binaries:ic-btc-adapter": "/opt/ic/bin/ic-btc-adapter:0755",  # Need for the Bitcoin integration.
+            "//rs/https_outcalls/adapter:ic-https-outcalls-adapter": "/opt/ic/bin/ic-https-outcalls-adapter:0755",  # Need for the HTTPS outcalls feature. `//publish/binaries:ic-https-outcalls-adapter` is for testing and must NOT be used here
+            "//publish/binaries:ic-crypto-csp": "/opt/ic/bin/ic-crypto-csp:0755",  # Crypto operations provider, required by the IC protocol (signing, etc).
+            "//publish/binaries:orchestrator": "/opt/ic/bin/orchestrator:0755",  # Replica process manager, required by the IC protocol (upgrades, node addition, etc).
+            ("//publish/malicious:replica" if malicious else "//publish/binaries:replica"): "/opt/ic/bin/replica:0755",  # Main protocol binary, required by the IC protocol. Installs the malicious replica if set.
```
Comment (nabdullindfinity) [inline ic-os/guestos/defs.bzl:38]: Could you please add a couple words about how "malicious" replica is only used in test builds?

File: ic-os/guestos/defs.bzl [hunk 7]
```diff
@@ -28,29 +28,29 @@ def image_deps(mode, malicious = False):
...
+            "//publish/binaries:ic-boundary-tls": "/opt/ic/bin/ic-boundary:0755",  # API boundary node binary, required by the IC protocol. The same GuestOS is used both for the replica and API boundary nodes.
+            "//publish/binaries:ic-consensus-pool-util": "/opt/ic/bin/ic-consensus-pool-util:0755",  # May be used during recoveries to export/import consensus pool artifacts.
+            "//publish/binaries:ic-regedit": "/opt/ic/bin/ic-regedit:0755",  # May be used for inspecting and recovering the registry.
+            "//publish/binaries:state-tool": "/opt/ic/bin/state-tool:0755",  # May be used during recoveries for calculating the state hash and inspecting the state more generally.
+            "//publish/binaries:ic-recovery": "/opt/ic/bin/ic-recovery:0755",  # Required for performing subnet recoveries on the node directly.
+            "//publish/binaries:fstrim_tool": "/opt/ic/bin/fstrim_tool:0755",  # TODO crypto, node
+            "//publish/binaries:guestos_tool": "/opt/ic/bin/guestos_tool:0755",  # TODO node
```
Comment (nabdullindfinity) [inline ic-os/guestos/defs.bzl:45]:
```suggestion
            "//publish/binaries:guestos_tool": "/opt/ic/bin/guestos_tool:0755",  # Tool for generating network config and hardware observability
```

File: ic-os/guestos/defs.bzl [hunk 8]
```diff
@@ -28,29 +28,29 @@ def image_deps(mode, malicious = False):
...
+            "//publish/binaries:nft-exporter": "/opt/ic/bin/nft-exporter:0755",  # TODO node
+            "//publish/binaries:vsock_guest": "/opt/ic/bin/vsock_guest:0755",  # TODO node
```
Comment (nabdullindfinity) [inline ic-os/guestos/defs.bzl:47]:
```suggestion
            "//publish/binaries:vsock_guest": "/opt/ic/bin/vsock_guest:0755",  # HostOS <--> GuestOS communication client
```

File: ic-os/guestos/defs.bzl [hunk 9]
```diff
@@ -28,29 +28,29 @@ def image_deps(mode, malicious = False):
...
+            "//cpp:infogetty": "/opt/ic/bin/infogetty:0755",  # TODO node
```
Comment (nabdullindfinity) [inline ic-os/guestos/defs.bzl:48]:
```suggestion
            "//cpp:infogetty": "/opt/ic/bin/infogetty:0755",  # Terminal manager that replaces the login shell
```

File: ic-os/guestos/defs.bzl [hunk 10]
```diff
@@ -28,29 +28,29 @@ def image_deps(mode, malicious = False):
...
+            "//cpp:prestorecon": "/opt/ic/bin/prestorecon:0755",  # TODO node
```
Comment (nabdullindfinity) [inline ic-os/guestos/defs.bzl:49]:
```suggestion
            "//cpp:prestorecon": "/opt/ic/bin/prestorecon:0755",  # Parallel restorecon replacement for filesystem relabeling
```

File: ic-os/guestos/defs.bzl [hunk 11]
```diff
@@ -28,29 +28,29 @@ def image_deps(mode, malicious = False):
...
+            "//publish/binaries:ic-boundary-tls": "/opt/ic/bin/ic-boundary:0755",  # API boundary node binary, required by the IC protocol. The same GuestOS is used both for the replica and API boundary nodes.
+            "//publish/binaries:ic-consensus-pool-util": "/opt/ic/bin/ic-consensus-pool-util:0755",  # May be used during recoveries to export/import consensus pool artifacts.
+            "//publish/binaries:ic-regedit": "/opt/ic/bin/ic-regedit:0755",  # May be used for inspecting and recovering the registry.
+            "//publish/binaries:state-tool": "/opt/ic/bin/state-tool:0755",  # May be used during recoveries for calculating the state hash and inspecting the state more generally.
+            "//publish/binaries:ic-recovery": "/opt/ic/bin/ic-recovery:0755",  # Required for performing subnet recoveries on the node directly.
+            "//publish/binaries:fstrim_tool": "/opt/ic/bin/fstrim_tool:0755",  # TODO crypto, node
+            "//publish/binaries:guestos_tool": "/opt/ic/bin/guestos_tool:0755",  # TODO node
+            "//publish/binaries:nft-exporter": "/opt/ic/bin/nft-exporter:0755",  # TODO node
```
Comment (nabdullindfinity) [inline ic-os/guestos/defs.bzl:46]:
```suggestion
            "//publish/binaries:nft-exporter": "/opt/ic/bin/nft-exporter:0755",  # Firewall (NFTables) counter exporter for observability
```

**Review by nabdullindfinity — APPROVED** [review]
Thank you!

**Pull request merged**